src/pages/ImportKey.vue

```diff
@@ -9,58 +9,71 @@ q-page(padding)
       outlined
       dense
       class="reward-address"
-      v-model="store.rewardAddress"
+      v-model="rewardAddress"
       input-class="text-center"
-      :rules="[val => isValidSubstrateAddress(val) || $t('importKey.addressErrorMsg')]"
+      :rules="[isValidSubstrateAddress]"
     )
   .row.justify-center.q-mt-sm
   .row.justify-end.items-center.q-mt-lg.absolute-bottom.q-pa-lg
     .col-auto.q-mr-md
       q-btn(
-        @click="$router.replace({ name: 'index' })" 
-        :label="$t('importKey.cancel')" 
-        outline 
-        size="lg" 
+        @click="$router.replace({ name: 'index' })"
+        :label="$t('importKey.cancel')"
+        outline
+        size="lg"
         icon-right="cancel"
       )
     q-space
     .col-auto
       q-btn(
-        :disable="!isValidSubstrateAddress(store.rewardAddress)"
+        :disable="!validAddress"
         :label="$t('importKey.continue')"
         @click="importKey()"
         icon-right="arrow_forward"
         outline
         size="lg"
       )
-      q-tooltip.q-pa-md(v-if="!isValidSubstrateAddress(store.rewardAddress)")
+      q-tooltip.q-pa-md(v-if="!validAddress")
         p.q-mb-lg {{ $t('importKey.tooltip') }}
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
-import { decodeAddress, encodeAddress } from '@polkadot/keyring';
-import { hexToU8a, isHex } from '@polkadot/util';
-
+import * as tauri from '@tauri-apps/api';
 import { useStore } from '../stores/store';
+import { errorLogger } from '../lib/util';
 
 export default defineComponent({
   setup() {
     const store = useStore();
     return { store };
   },
+  data() {
+    return {
+      rewardAddress: "",
+      validAddress: false,
+    };
+  },
   methods: {
-    isValidSubstrateAddress(val: string): boolean {
+    async isValidSubstrateAddress(addr: string): Promise<boolean|string> {
       try {
-        encodeAddress(isHex(val) ? hexToU8a(val) : decodeAddress(val));
-        return true;
+        let result: boolean = await tauri.invoke("validate_reward_address", { addr });
+        this.validAddress = result;
+        return (result || this.$t('importKey.rewardAddress'));
       } catch (error) {
-        return false;
+        errorLogger(error);
+        return (false || this.$t('importKey.rewardAddress'));
       }
     },
     async importKey() {
+      this.store.rewardAddress = this.rewardAddress
```

Comment:
it's better to use a method from `store` (smth like `setRewardAddress`) instead of assignment